cpp/arcticdb/processing/clause.cpp

```diff
@@ -658,4 +658,107 @@ std::string DateRangeClause::to_string() const {
+                                        top_k.insert(NamedColumnSortedByDistance(
+                                                sum_differences_exponentiated,
+                                                column.element,
+                                                s.segment(store).field(column.index).name()));
+
+
+                                        if (top_k.size() > k) {
+                                            top_k.erase(*top_k.rbegin());
+                                        }
+                                    }
+                                });
+                            }
+                        }
+                        );
+            }
+            );
+    // By this time we should have iterated through each block of each SegmentInMemory of each ProcessingSegment;
+    // and in each case read a vector and inserted it if more similar than the running top_k. Now we write to a new
+    // segment.
+    SegmentInMemory seg;
+    seg.descriptor().set_index(IndexDescriptor(0, IndexDescriptor::ROWCOUNT));
+    seg.set_row_id(query_vector.size() - 1 + 1);
+    // -1 because of zero-indexing, +1 because we want to add the distance on at the end.
+    for (const auto& column : top_k) {
+        // What's below is the natural way of doing things. However:
+        // it writes non-contiguous blocks. When the result of a query is read
+        // we use a read method that expects a pointer to a contiguous block
+        // rather than something else that will put non-contiguous blocks onto memory.
+        // Instead we make a copy.
+//        seg.add_column(scalar_field(DataType::FLOAT64,
+//                                    column.name),
+//                       column.column);
+        column.column->type().visit_tag([&, lp] (auto type_desc_tag) {
+            using TypeDescriptorTag = decltype(type_desc_tag);
+            using RawType = typename TypeDescriptorTag::DataTypeTag::raw_type;
+
+            auto write_col = std::make_shared<Column>(column.column->type(), query_vector.size()+1, true, false);
+
+
+            auto col_data = column.column->data();
+            auto write_ptr = reinterpret_cast<RawType *>(write_col->ptr());
+            while (auto block = col_data.next<TypeDescriptorTag>()) {
+                auto ptr = reinterpret_cast<const RawType *>(block.value().data());
+                for (auto i = 0u; i < block.value().row_count(); ++i, ++ptr) {
+                    *write_ptr = RawType(*ptr);
+                    ++write_ptr;
+                }
+            }
+            *write_ptr = RawType(pow(column.distance, 1/static_cast<float>(lp)));
+            write_col->set_row_data(query_vector.size());
+            seg.add_column(scalar_field(DataType::FLOAT64, column.name), write_col);
+        });
+        }
+    Composite<ProcessingSegment> output;
+    ProcessingSegment segment = ProcessingSegment{std::move(seg)};
+    auto descriptor = segment.data().at(0); // now empty
```

Comment:
Dead code?

Reply:
Yes, debugging stuff; sorry, should've caught this.